### PR TITLE
fix: remove deprecated GET /releases endpoint

### PR DESCRIPTION
**Pull request by kjaskiewiczz**
Changelog: Title
Ticket: MEN-5180

BREAKING CHANGE: remove deprecated GET /releases endpoint

### CODE REVIEW

**Comment by kjaskiewiczz** [comment]
@mender-test-bot start pipeline

**Review by merlin-northern — APPROVED** [review]
lgtm, as a review I checked if there are any leftovers; looks ok.
is the fact that we have `/deployments/releases/list` the reason for removing the endpoint in this PR? do we want to provide some fall-through to the former to preserve a bit of the API?

**Review by merlin-northern — APPROVED** [review]
lgtm, as a review I checked if there are any leftovers; found only [this](https://github.com/mendersoftware/deployments/pull/863#discussion_r1191955846). looks ok.
is the fact that we have `/deployments/releases/list` the reason for removing the endpoint in this PR? do we want to provide some fall-through to the former to preserve a bit of the API?

File: api/http/releases_test.go
Comment (merlin-northern) [inline api/http/releases_test.go:1]: could you check `TestGetReleasesFilter` there is `reqUrl := "http://1.2.3.4/api/management/v1/deployments/releases"`

**Review by alfrunes — APPROVED** [review]

**Pull request closed without merging**